### PR TITLE
Fix performance issue

### PR DESCRIPTION
**Pull request by ghost**
- Simplified code to Analyse the emf resources and the model
- removed crossreferencer to detect dangling references. was too expensive calculation wise.
- added code which checks all crossreferences from a resource if they are contained instead.
- added a new test case for it
- Adjusted editor to also show emf warnings and not just errors as primary page.

### CODE REVIEW

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java [hunk 1]
```diff
@@ -1187,34 +1137,41 @@ public Resource createResource(URI uri) {
...
+			// In case there are null objects which should not be in the resource
+			// something went seriously wrong. No other analysis has to be executed
+			// If there are no null diagnostics, continue with the other checks
+			if (resourceDiagnostic.getSeverity() == Diagnostic.OK) {
+				BasicDiagnostic modelDiagnostic = analyzeModelProblems(resource);
+				resourceDiagnostic.merge(modelDiagnostic);
+			}
+
+			// Now check severity levels, and set the diagnostics map accordingly
+			// The map will be sued by the editors to display the error messages etc.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java]: sued -> used

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java [hunk 2]
```diff
@@ -1224,61 +1181,101 @@ public boolean updateDiagnostic(Resource resource) {
...
 			}
 		}
-
-		return resourceDiagnostics;
+		
+		return returnDiagnostic;
+	}
+	
+	/**
+	 * This method create EMF Diagnostics on the Model Object in case of detected problems
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java]: create -> creates

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java [hunk 3]
```diff
@@ -1224,61 +1181,101 @@ public boolean updateDiagnostic(Resource resource) {
...
+				// Check all cross references
+				eObject.eCrossReferences().forEach((eReferencedObject) -> {
+					// Check containment of all other objects
+					if (eReferencedObject.eResource() == null) {
+						returnDiagnostic.merge(new BasicDiagnostic(
+								Diagnostic.WARNING,
+								Activator.getPluginId(),
+								0,
+								"Found a dangling reference in due to uncontained object. Press save to fix.",
+								new Object[] { eReferencedObject, eObject }
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java]: `in due` sentence seems unfinished.

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/GenericEditor.java [hunk 4]
```diff
@@ -587,7 +587,7 @@ protected void addPages() {
 		}
 
 		Diagnostic diagnostic = editingDomain.getResourceSet().getResourceToDiagnosticsMap().get(resource);
-		boolean hasErrors = diagnostic != null && diagnostic.getSeverity() == Diagnostic.ERROR;
+		boolean hasErrors = diagnostic != null && diagnostic.getSeverity() != Diagnostic.OK;
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/GenericEditor.java]: Just for clarity: According to the new code I think we would stay on the error page in the case of a warning.
Is this intended?

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
Mostly just minor typos in descriptions.

**Review by franzTobiasDLR — APPROVED** [review]

File: de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/GenericEditor.java (hunk 4 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/GenericEditor.java:590]: Yes this is intended. We want to react to warnings as well. It is also not staying on the page, but prominently showing it.

Comment (SaMuellerDLR) [reply de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/GenericEditor.java:590]: Alright

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Comment by PhilMFischer** [comment]
Closes #509

**Pull request merged**